upgrade-2.x.sh

```diff
@@ -1178,12 +1173,21 @@ if version_gt "${VERSION_ID}" "${minimum_hostapp_target_version}" ||
     [ "${VERSION_ID}" == "${minimum_hostapp_target_version}" ]; then
     log "hostapp-update command exists, use that for update"
     progress 50 "Running OS update"
-    if [ -n "${delta_image}" ] && hostapp_based_update "${delta_image}"; then
-        # overriding this for the later supervisor version inspection, since we've updated successfully
-        image=${delta_image}
-    else
-        log "Delta failed or not found, falling back to regular pull"
-        hostapp_based_update "${image}" || log ERROR "hostapp-update has failed..."
+    images=("${delta_image}" "$(get_image_location "${target_version/_/+}")" "${upstream_image}")
+    image_types=("delta" "balena_registry" "upstream_registry")
```

Comment:
It looks like you're essentially building an associative array.  Would it be worth using that Bash feature directly?  I'm not a fan of the Bash syntax for arrays (linear or associative), so if you want to skip it, it would be worth adding a comment to make the purpose clear.

Reply:
Shall add a comment, i really don't care for the associative arrays in bash!